internal/kafka/internal/handlers/validation.go

```diff
@@ -199,12 +212,14 @@ func getCloudProviderAndRegion(
 	providerName := arrays.FirstNonEmptyOrDefault(defaultProvider.Name, kafkaRequest.CloudProvider)
 	// Validation for Cloud Provider
 	provider, providerSupported := supportedProviders.GetByName(providerName)
-	if !providerSupported {
+
+	// We only return a validation error if the provider is not supported and Kafka doesn't have a dedicated data plane cluster assigned to it
+	if !providerSupported && shared.StringEmpty(kafkaRequest.ClusterId) {
```

Comment:
should we add an additional check for instance type == enterprise here? Cluster id is (currently at least) only supported for enterprise, in all other cases the normal validation should take place.

Reply:
On second thought, that makes the condition somewhat hard to reason about. What do you think?

Reply:
That's already done here in the `validateKafkaBillingModel` validator